yt_idefix/data_structures.py

```diff
@@ -430,6 +458,270 @@ def _get_code_version(self) -> str:
         return match.group()
 
 
+class PlutoStaticDataset(IdefixDataset):
+    """A common abstraction for PlutoXdmfDataset and PlutoVtkDataset."""
+
+    _version_regexp = re.compile(r"v\d+\.\d+\.?\d*[-\w+]*")
+    _dataset_type: str  # defined in subclasses
+    _required_header_keyword = "PLUTO"
+    _default_definitions_header = "definitions.h"
+    _default_inifile = "pluto.ini"
+
+    def __init__(
+        self,
+        filename,
+        *,
+        dataset_type: str | None = None,  # deleguated to child classes
+        units_override: dict[str, UnitLike] | None = None,
+        unit_system: Literal["cgs", "mks", "code"] = "cgs",
+        default_species_fields: Literal["neutral", "ionized"] | None = None,
+        # from here, frontend-specific arguments
+        geometry: Literal["cartesian", "spherical", "cylindrical", "polar"]
+        | None = None,
+        inifile: str | os.PathLike[str] | None = None,
+        definitions_header: str | os.PathLike[str] | None = None,
+        parameters: dict[str, Any] | None = None,
```

Comment:
I would rather have exposed parameters exposed directly.
```suggestion
        gamma: float | None = None,
        mu: float | None = None,
```

Reply:
Okay!